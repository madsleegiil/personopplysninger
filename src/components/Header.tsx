import * as React from 'react';

export interface Props {
    name: string;
}

const Header = (props: {name: string}) => {
    return (
        <header className="header">
            <h1 className="title">Dine personopplysninger</h1>
            <div>Hei {props.name}, her kan du se og laste ned informasjon NAV har nyttet til deg.</div>
            <div>Vi har samlet informasjonen vi har tilgjengelig knyttet til deg p√• denne siden.</div>
            <div>Les om personvern i Arbeids- og velferdsetaten her.</div>
        </header>
    );
};

export default Header;
