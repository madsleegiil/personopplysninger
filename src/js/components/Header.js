import React, { Component } from 'react';

class Header extends Component {
  render() {
    return (
      <div className="Header">
        <h1 align="center">Dine personopplysninger</h1>
      </div>
    );
  }
}

export default Header;
