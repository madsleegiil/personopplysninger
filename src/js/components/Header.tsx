import React from "react";
import { FormattedHTMLMessage } from "react-intl";
import { Ingress, Innholdstittel } from "nav-frontend-typografi";
import group from "../../assets/img/group.svg";
import veileder from "../../assets/img/veileder.svg";

const Header = (props: any) => {
  const { fornavn } = props;
  return (
    <div className="header">
      <div className="header__content">
        <Innholdstittel>
          <FormattedHTMLMessage id="header.hello" values={{ name: fornavn }} />
        </Innholdstittel>
        <div className="header__content-intro">
          <Ingress>
            <FormattedHTMLMessage
              id="header.intro"
              values={{ name: fornavn }}
            />
          </Ingress>
        </div>
        <div className="header__content-description">
          <Ingress>
            <FormattedHTMLMessage id="header.description" />
          </Ingress>
        </div>
      </div>
      <img src={group} className="header__icon-desktop" alt="Pc" />
      <div className="header__icon-mobile-container">
        <img src={veileder} className="header__icon-mobile" alt="Pc" />
      </div>
    </div>
  );
};

export default Header;
