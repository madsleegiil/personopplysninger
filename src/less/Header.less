.header {
  background-color: #DADD9C;
  height: 20rem;

  .title {
    margin-top: 4rem;
    font-size: 3rem;
  }
}
