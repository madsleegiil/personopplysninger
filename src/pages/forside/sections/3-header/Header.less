.header {
  &__seksjon {
    padding: 0.5rem 0;
  }
  &__ikon {
    width: 100% !important;
    height: 100% !important;
  }
  &__content {
    margin-bottom: 0.5rem;
  }
  @media print {
    display: none;
  }
}
