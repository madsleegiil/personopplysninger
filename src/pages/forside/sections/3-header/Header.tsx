import React from "react";
import { FormattedMessage } from "react-intl";
import { Normaltekst, Systemtittel } from "nav-frontend-typografi";
import veilederIkon from "assets/img/Veileder.svg";
import Veilederpanel from "nav-frontend-veilederpanel";
import Error from "components/error/Error";
import Spinner from "../4-personinfo/PersonInfo";
import { formatName } from "utils/text";
import { useStore } from "store/Context";
import Lenke from "nav-frontend-lenker";

const Header = () => {
  const [{ nameInfo }] = useStore();

  switch (nameInfo.status) {
    default:
    case "LOADING":
      return <Spinner />;
    case "RESULT":
      const { name } = nameInfo.data;
      const fornavn = name.split(" ")[0];
      const Veileder = (
        <img src={veilederIkon} className="header__ikon" alt="Veileder" />
      );

      return (
        <div className="header">
          <Veilederpanel svg={Veileder} type={"plakat"} kompakt={true}>
            <div className="box__container header__content">
              <Systemtittel>
                {fornavn ? (
                  <FormattedMessage
                    id="header.hello.name"
                    values={{ name: formatName(fornavn) }}
                  />
                ) : (
                  <FormattedMessage id="header.hello" />
                )}
              </Systemtittel>
              <div className="header__seksjon">
                <Normaltekst>
                  <FormattedMessage id="header.obs" />
                </Normaltekst>
              </div>
              <div className="header__seksjon">
                <Normaltekst>
                  <FormattedMessage
                    id="header.description"
                    values={{
                      a: (text: String) => (
                        <Lenke
                          href="/personvern"
                          target="blank"
                          rel="noopener noreferrer"
                        >
                          {text}
                        </Lenke>
                      ),
                    }}
                  />
                </Normaltekst>
              </div>
            </div>
          </Veilederpanel>
        </div>
      );
    case "ERROR":
      return <Error error={nameInfo.error} />;
  }
};
export default Header;
