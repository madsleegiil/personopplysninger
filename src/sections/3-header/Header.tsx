import React from "react";
import { FormattedHTMLMessage } from "react-intl";
import { Ingress, Innholdstittel } from "nav-frontend-typografi";
import group from "../../assets/img/group.svg";
import veileder from "../../assets/img/veileder.svg";

interface Props {
  fornavn?: string;
}

const Header = ({ fornavn }: Props) => (
  <div className="header">
    <div className="header__content">
      <Innholdstittel>
        <FormattedHTMLMessage
          id="header.hello"
          values={{ name: fornavn || "" }}
        />
      </Innholdstittel>
      <div className="header__content-intro">
        <Ingress>
          <FormattedHTMLMessage
            id="header.intro"
            values={{ name: fornavn || "" }}
          />
        </Ingress>
      </div>
      <div className="header__content-description">
        <Ingress>
          <FormattedHTMLMessage id="header.description" />
        </Ingress>
      </div>
    </div>
    <img src={group} className="header__icon-desktop" alt="PC" />
    <div className="header__icon-mobile-container">
      <img src={veileder} className="header__icon-mobile" alt="PC" />
    </div>
  </div>
);
export default Header;
