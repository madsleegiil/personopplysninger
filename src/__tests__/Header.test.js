import * as React from 'react';
import Header from 'js/components/Header';
import wrapIntl from 'js/IntlTestHelper';
const ReactTestRenderer = require('react-test-renderer');

test('render Header with content', () => {
  const component = ReactTestRenderer.create(wrapIntl(<Header
    fornavn="Donald"
  />));
  expect(component.toJSON()).toMatchSnapshot();
});
