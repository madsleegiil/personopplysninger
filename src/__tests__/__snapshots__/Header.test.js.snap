// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Header with content 1`] = `
<div
  className="header"
>
  <div
    className="header__content"
  >
    <h1
      className="typo-innholdstittel"
    >
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "Hei Donald!",
          }
        }
      />
    </h1>
    <div
      className="header__content-intro"
    >
      <p
        className="typo-ingress"
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "Her kan du se opplysninger som NAV har registrert om deg.",
            }
          }
        />
      </p>
    </div>
    <div
      className="header__content-description"
    >
      <p
        className="typo-ingress"
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "<a href='https://www.nav.no/personvern' target='blank' rel='noopener noreferrer' class='lenke'>Les om personvern i Arbeids- og velferdsetaten her.</a>",
            }
          }
        />
      </p>
    </div>
  </div>
  <img
    alt="Pc"
    className="header__icon-desktop"
    src="group.svg"
  />
  <div
    className="header__icon-mobile-container"
  >
    <div
      className="header__icon-mobile-circle"
    >
      <img
        alt="Pc"
        className="header__icon-mobile"
        src="veilder.svg"
      />
    </div>
  </div>
</div>
`;
